File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | distanceFromOrigin > 10 = 0
  | distanceFromOrigin > 5 = 1
  | distanceFromOrigin > 1 = 5
  | otherwise = 10
  where
    distanceFromOrigin = sqrt (x * x + y * y)

File: solutions/ocaml/darts/1/darts.ml
let score (x : float) (y : float) : int =
  let distance = sqrt ((x ** 2.) +. (y ** 2.)) in
  match distance with
  | x when x <= 1.0 -> 10
  | x when x <= 5.0 -> 5
  | x when x <= 10.0 -> 1
  | _ -> 0
